### PR TITLE
Removing "netcoreapp"  and "netfx" TargetGroup

### PR DESCRIPTION
**Pull request by Anipik**
Most of the work is done using regex. The commit messages contain the regex used for that commit.
I also introduced a new variable to track the latest version of netcoreapp, making it easier to increment for later version
.sln files will need to be updated with every new major product version.

Regexs used
netcoreapp -> netcoreapp5.0 in .sln
netcoreapp; -> netcoreapp5.0; in .props
netcoreapp- -> netcoreapp5.0- in .props
netcoreapp- -> netcoreapp5.0- in .csproj
'netcoreapp -> 'netcoreapp5.0 in .csproj

netcoreapp5.0 occurences are then replaced using the above regex by TargetFrameworkVNext

Same stuff has been done for netfx

### CODE REVIEW

**Comment by ViktorHofer** [comment]
This collides with the work being done in https://github.com/dotnet/runtime/pull/445. Can you either pause this until the other PR is in or use the same TFM name so that we can simply replace the property define here after the other one is in?

**Comment by Anipik** [comment]
Sure I can wait until the #445 is in

File: Directory.Build.props
Comment (jkotas) [inline Directory.Build.props]: Should this rather be `DefaultTargetFramework` or `CurrentTargetFramework`?

VNext will be very confusing in release/servicing branches or in the source build.

Comment (ericstj) [reply Directory.Build.props]: Agreed.  Perhaps `$(netcoreappCurrent)` so that it parses well when seen in `<TargetFrameworks>`

File: src/libraries/sendtohelix.proj
Comment (safern) [inline src/libraries/sendtohelix.proj]: This change is not correct from my understanding. We're using the `BuildConfiguration` that is calculated in `Directory.Build.targets` which contains OSGroup, Configuration and Arch

**Comment by ViktorHofer** [comment]
We agreed on `NetCoreAppCurrentTargetFrameworkMoniker` (.NETCoreApp,Version=v5.0) and `NetCoreAppCurrent` (netcoreapp5.0) in https://github.com/dotnet/runtime/pull/445.

**Comment by ViktorHofer** [comment]
@Anipik this work is now unblocked.

**Comment by Anipik** [comment]
@ericstj @ViktorHofer @safern  can you please take a look ?

**Comment by Anipik** [comment]
> Shouldn't we have a property for NetFrameworkCurrent as well?

we needed the NetCoreAppCurrent property so that we dont have to add a new targetframework when we move to higher versions. This wont be the case with full framework

**Comment by ericstj** [comment]
>  This wont be the case with full framework

Should we update to [net48](https://devblogs.microsoft.com/dotnet/announcing-the-net-framework-4-8/) then?

File: eng/build.ps1
Comment (ericstj) [inline eng/build.ps1]: Hmm, I'm not a fan of hardcoding the mapping here.  Can we handle this differently?

File: eng/restore/docs.targets
Comment (ericstj) [inline eng/restore/docs.targets]: Can this be changed to use $(XmlDocFileRoot)?

**Comment by ViktorHofer** [comment]
> Should we update to net48 then?

Is there a business / infrastructure justification for doing that? There was no need to upgrade in the last year.

**Comment by ericstj** [comment]
I don't recall explicitly considering it since it was released.  We held ourselves to 4.6.1 for a while when we were working on netstandard2.0, then we updated to 4.7.2 because that was the latest at the time.  I would presume we'd do one last update to 4.8 to be on the latest and then never touch it again.  I'm OK with not doing it now: if we do delay then I'd agree with you to have an indirection property to avoid churning twice.

**Review by ericstj — COMMENTED** [review]
This appears to have broken category filtering of tests (thus all the failures).  Trying to hunt down where you need to fix that.

**Comment by ViktorHofer** [comment]
Afaik we upgraded from net461 to net472 primarily because it supported netstandard inbox without some of binding redirects issues, but I could be wrong here.

> if we do delay then I'd agree with you to have an indirection property to avoid churning twice.

To be honest, as we don't have that many .NET Framework projects anymore (wasn't it around 50?) I'm fine with not introducing a property as a) the indirection property makes it harder to read and b) we will never touch that property anytime again after the upgrade. That said, I don't have a strong opinion here.

File: eng/build.ps1
Comment (Anipik) [reply eng/build.ps1]: This is just a temporary fix until the yaml files are finalized, then i go ahead and delete this. I can add a comment here to remove the stuff

File: eng/restore/docs.targets
Comment (ViktorHofer) [reply eng/restore/docs.targets]: Actually this change is interesting. You are assuming that the directory name in the intellisense package changed from netcoreapp to netcoreapp5.0. Is this true?

File: eng/build.ps1
Comment (ViktorHofer) [reply eng/build.ps1]: Are you delaying the yaml change to avoid git conflicts?

**Comment by ericstj** [comment]
Here's the test piece that's broken: https://github.com/dotnet/runtime/blob/1924be9b263d1e416717166d952d3a3d16c963e1/eng/testing/xunit/xunit.targets#L19

I think you need to define that category different.

File: eng/restore/docs.targets
Comment (ericstj) [reply eng/restore/docs.targets]: @ViktorHofer is right, I was assuming that we'd want to change this in step with TFM but its probably better to just keep it generic (or even remove the folder).

**Comment by ViktorHofer** [comment]
> The whole TFM filtering system seems rather fragile: can't we just tell the runner what the target framework is (or have it detect it) then filter that way? Why even bother flowing through categories

We tell the runner the target framework (at least for VSTest which also auto-detects it by looking into the test assembly's attributes). These categories are to filter tests with traits (xunit) / categories (vstest). I suggest to either use the versionless part of the TargetFramework property or condition based on the TargetFrameworkIdentifier (with its helpers in the runtime repo, ie `TargetsNetCoreApp`).

**Comment by ViktorHofer** [comment]
cc @safern

File: eng/build.ps1
Comment (Anipik) [reply eng/build.ps1]: yes

File: eng/restore/docs.targets
Comment (safern) [reply eng/restore/docs.targets]: cc: @carlossanlop @ahsonkhan we generate this package manually. So if this is going to change we just need to be aware so that we update the folder in the package. 

Also, the docs team is going to be generating a NuGet package for us in the feature.

I would leave it as `netcoreapp`.

Comment (ahsonkhan) [reply eng/restore/docs.targets]: > I would leave it as `netcoreapp`.

Yep, agreed.

**Comment by ericstj** [comment]
Here's another place that requires fixup: https://github.com/dotnet/runtime/blob/73eec1b1223eb5d974a3ceff2419f2dc425ae4d1/src/coreclr/tests/scripts/patch-corefx.py#L248

**Comment by ericstj** [comment]
New failure introduced by bad versionless path here: https://github.com/dotnet/runtime/blame/6b7890626d6a5f7c8db1649dcde3761b7ff93455/eng/pipelines/libraries/build-job.yml#L115-L118

File: eng/pipelines/libraries/build-job.yml
Comment (safern) [inline eng/pipelines/libraries/build-job.yml]: FYI: @jkoritzinsky

File: src/libraries/Microsoft.VisualBasic.Core/src/Microsoft.VisualBasic.Core.vbproj
Comment (safern) [inline src/libraries/Microsoft.VisualBasic.Core/src/Microsoft.VisualBasic.Core.vbproj]: Why isn't this one using `$(NetCoreAppCurrent)`

File: src/libraries/Native/native-binplace.proj
Comment (safern) [inline src/libraries/Native/native-binplace.proj]: Should this one use `NetCoreAppCurrent`? Or this doesn't import the root `Directory.Build.props`?

Comment (safern) [reply src/libraries/Native/native-binplace.proj]: If so, I believe we should use it to mitigate errors when we move to the next version of netcoreapp.

File: src/libraries/System.Data.SqlClient/tests/StressTests/System.Data.SqlClient.Stress.Tests/System.Data.SqlClient.Stress.Tests.csproj
Comment (safern) [inline src/libraries/System.Data.SqlClient/tests/StressTests/System.Data.SqlClient.Stress.Tests/System.Data.SqlClient.Stress.Tests.csproj]: I believe we can remove this property, we now use `IgnoreForCI`, right, @ViktorHofer ?

Comment (ViktorHofer) [reply src/libraries/System.Data.SqlClient/tests/StressTests/System.Data.SqlClient.Stress.Tests/System.Data.SqlClient.Stress.Tests.csproj]: Weird that I didn't capture this case with my change. Yes please use [IgnoreForCI](https://github.com/dotnet/runtime/blob/master/eng/testing/tests.targets#L23)

File: eng/packaging.props
Comment (ViktorHofer) [inline eng/packaging.props]: another case where you should keep netcoreapp?

File: src/libraries/Directory.Build.props
Comment (ViktorHofer) [inline src/libraries/Directory.Build.props]: This should use the current netcoreapp property.

File: src/installer/pkg/projects/netcoreapp/src/localnetcoreapp.override.targets
Comment (safern) [inline src/installer/pkg/projects/netcoreapp/src/localnetcoreapp.override.targets]: Why not use $(NetCoreAppCurrent) here? Is it not defined here?

Comment (Anipik) [reply src/installer/pkg/projects/netcoreapp/src/localnetcoreapp.override.targets]: Its defined, I thought it wasn't. I can change that in a follow up PR (Just want to get this in before more merge conflicts)

**Comment by jkoritzinsky** [comment]
I'd prefer if I could get #494 in before this if I can to unblock cross-cutting PRs instead of having to react to this and delaying merging #494 another day or so.

**Comment by safern** [comment]
> I'd prefer if I could get #494 in before this if I can to unblock cross-cutting PRs

I definitely agree #494 has higher priority since that unblocks live live workflow.

**Comment by Anipik** [comment]
sure  i can wait. let me know when i can go ahead and merge this one.

File: Directory.Build.props
Comment (ViktorHofer) [inline Directory.Build.props]: You are duplicating this property which already exists in L33: https://github.com/dotnet/runtime/pull/457/files#diff-4b2f402501d23abbede6eac0f47783e4R33.

File: src/libraries/Directory.Build.props
Comment (ViktorHofer) [inline src/libraries/Directory.Build.props]: Shouldn't this be $(NetCoreAppCurrent) as well? Have you checked other places where you are hardcoding netcoreapp5.0?

File: src/libraries/Directory.Build.targets
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets]: Here and in many other places you are hardcoding netcoreapp5.0. This is very unfortunate as we will bump the tfm at least every year going forward. Can you please update those instances as well to use the property?

File: src/installer/pkg/projects/netcoreapp/src/localnetcoreapp.override.targets
Comment (ViktorHofer) [reply src/installer/pkg/projects/netcoreapp/src/localnetcoreapp.override.targets]: There is still more work required here, I would make the change as part of this PR.

**Comment by ViktorHofer** [comment]
Actually I just saw in how many places we hardcode net472 and I'm not really happy about that. As we are already touching so many files as part of this PR I would introduce a property for it `NetFrameworkCurrent` as well as part of this PR. As this PR is currently blocked by other work we should have some time for it. I can help too.

**Comment by Anipik** [comment]
@ViktorHofer as we are not gonna increase the netframework version, what will be the advantages of having this new property ?

**Comment by Anipik** [comment]
@jkoritzinsky @safern @ViktorHofer 
Do we need to add build coreclr step for the runtime-libraries enterprise-linux ?
```
  /repo/eng/liveBuilds.targets(20,5): error : The CoreCLR artifacts path does not exist '/repo/artifacts/bin/coreclr/Linux.x64.Debug/'. The CoreCLR subset category must be built before building this project. [/repo/src/libraries/System.AppContext/src/System.AppContext.csproj]
/repo/eng/liveBuilds.targets(20,5): error : The CoreCLR artifacts path does not exist '/repo/artifacts/bin/coreclr/Linux.x64.Debug/'. The CoreCLR subset category must be built before building this project. [/repo/src/libraries/System.Buffers/src/System.Buffers.csproj]
/repo/eng/liveBuilds.targets(20,5): error : The CoreCLR artifacts path does not exist '/repo/artifacts/bin/coreclr/Linux.x64.Debug/'. The CoreCLR subset category must be built before building this project. [/repo/src/libraries/System.Private.Uri/src/System.Private.Uri.csproj]
/repo/eng/liveBuilds.targets(20,5): error : The CoreCLR artifacts path does not exist '/repo/artifacts/bin/coreclr/Linux.x64.Debug/'. The CoreCLR subset category must be built before building this project. [/repo/src/libraries/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj]
/repo/eng/liveBuilds.targets(20,5): error : The CoreCLR artifacts path does not exist '/repo/artifacts/bin/coreclr/Linux.x64.Debug/'. The CoreCLR subset category must be built before building this project. [/repo/src/libraries/System.Diagnostics.Contracts/src/System.Diagnostics.Contracts.csproj]
/rep
```

**Comment by jkoritzinsky** [comment]
Yes. You'll need to add a job to build coreclr for that pipeline now that live-live is enabled.

**Comment by Anipik** [comment]
@safern @ViktorHofer @ericstj can we go ahead and merge this one ? 
linux failure is not related to this one .

**Comment by davidsh** [comment]
>runtime-libraries enterprise-linux — #20191213.7 failed 

This has been fixed already with PR #843 which was merged in the last hour.

**Comment by Anipik** [comment]
Thanks @davidsh

**Comment by Anipik** [comment]
This one is ready to go in. the merging is blocked and require an approval

**Review by safern — APPROVED** [review]

**Pull request merged**
